src/lib.rs

```diff
@@ -94,7 +94,9 @@ fn compress<T: Clone + Copy + PartialEq>(
             } else {
                 d += 1;
                 let s = r.len();
-                r.resize(s + 1, empty_val); // increase result vector size
+                if d + row_length > s {
```

Comment:
Won't this always be true?

Reply:
I shouldn't, as `d` is reset to 0 for every row. Let's take as an example `[1,0,0,2,0,0,3,0,0]` with row-length 3.
At first we set `r`'s length to 3. Then we push the first row `[1,0,0]` onto it resulting in `r = [1,0,0]`.

Then we try to push the second row `[2,0,0]`. Since it doesn't fit, we increase `d` by 1. Since `d + row_length > r.len()`, we need to resize `r` by one (or to `d + row_length`). Afterwards we can apply the vector and `r = [1,2,0,0]`. 

Now we reset `d` and try to push the third row `[3,0,0]`. It doesn't fit into position 0. So we increase `d` by 1. However, since `r` is already at the correct size, this doesn't trigger a resize. However, the row still doesn't fit. So we increase `d` again by 1. This time `d + row_length > r.len()` so we need to resize `r` again. Afterwards, we can apply the row with a displacement of 2, resulting in `r = [1,2,3,0,0]`.

Reply:
Ah, I think I get it now